pinot-common/src/test/java/org/apache/pinot/common/function/JsonFunctionsTest.java

```diff
@@ -65,4 +65,42 @@ public void testJsonFunction()
     assertEquals(JsonFunctions.jsonPathDouble(jsonString, "$.actor.aaa", 53.2), 53.2);
   }
 
+  @Test
+  public void testJsonFunctionExtractingArray()
+      throws JsonProcessingException {
+    String jsonString = "{\n" +
+        "    \"name\": \"Pete\",\n" +
+        "    \"age\": 24,\n" +
+        "    \"subjects\": [\n" +
+        "        {\n" +
+        "            \"name\": \"maths\",\n" +
+        "        \"grade\": \"A\"\n" +
+        "        },\n" +
+        "        {\n" +
+        "            \"name\": \"maths\",\n" +
+        "        \"grade\": \"B\"\n" +
+        "        }\n" +
+        "    ]\n" +
+        "}";
+    assertEquals(JsonFunctions.jsonPathArray(jsonString, "$.subjects[*].name"), new String[]{"maths", "maths"});
+    assertEquals(JsonFunctions.jsonPathArray(jsonString, "$.subjects[*].grade"), new String[]{"A", "B"});
+  }
+
+  @Test
+  public void testJsonFunctionOnJsonArray()
+      throws JsonProcessingException {
+    String jsonArrayString =
+        "[\n" +
+        "        {\n" +
+        "            \"name\": \"maths\",\n" +
+        "        \"grade\": \"A\"\n" +
+        "        },\n" +
```

Comment:
Can we try nested array as well?
Something like subjects 
`{ name, array of lastTwoSemesterGrades}`

Reply:
Also, scalar functions should also have e2e query execution unit tests. Alternatively, may be add a query compilation unit test as well to ensure it gets recognized/compiled as scalar

Reply:
added.

Reply:
Should have mentioned earlier itself. Can we please add one test for empty array as well?

Also,  Can we chain the jsonPathArray with jsonExtract as a transform and extract a particular value out of the array? Or for such cases, the recommended way is to directly use jsonExtract as opposed to doing jsonExtract(jsonPathArray(....)) ?